examples/khr-3hv/khr-3hv_continuous/controllers/supervisorsManager/khr3hvEnv.py

```diff
@@ -0,0 +1,206 @@
+        # Shift the last 10 observations by one, on the time window of 5
+        self.obs = shift(self.obs, -self.numObs, cval=np.NaN)
+        self.obs[-self.numObs:] = motorPos
+
+        return np.array(self.obs)
+
+    def get_reward(self, action):
+        """
+        Calculate the reward of the agent. The reward award agent when moving forward on the x-axis.
+        Also, we are using a second term so the agent moves further away from his previous position.
+        :return: Reward value
+        :rtype: float
+
+        """
+        reward = 2.5 * \
+            self.robot.getPosition()[0] + \
+            self.robot.getPosition()[0] - self.prev_pos
+
+        if self.counterLoging % self.save_freq == 0 and self.train and not self.useRay:
+            wandb.log({"reward": reward,
+                       "reward-1term-weight-pos": 2.5 * self.robot.getPosition()[0],
+                       "reward-2term-diff-possition": self.robot.getPosition()[0] - self.prev_pos
+                       })
+
+        self.prev_pos = self.robot.getPosition()[0]
+        return reward
+
+    def is_done(self):
+        """
+        This method checks the termination criteria for each episode. 
+        If the criteria are satisfied return True otherwise it returns False.
+        :return: Termination criteria
+        :rtype: Boolean
+
+        """
+        # Agent fall.
+        if self.robot.getPosition()[2] < 0.5:
+            return True
+        # Agent walked out of the box
+        if self.robot.getPosition()[0] > 15:
+            return True
+        return False
+
+    def get_default_observation(self):
+        """
+        Simple implementation returning the default observation which is a zero 
+        vector in the shape of the observation space.
+        :return: Starting observation zero vector
+        :rtype: numpy array
+        """
+        self.prev_pos = 0
+        self.obs = np.zeros(self.t * self.numObs)
+        self.distance_walk = 0
+
+        return np.zeros(self.observation_space.shape[0])
+
+    def apply_action(self, action):
+        """
+        This method uses the action list provided, which contains the next action to be 
+        executed by the robot.It contains a float number denoting the action. 
+        The corresponding motorPosition value is applied at each motor.
+        :param action: The list that contains the action value
+        :type action: list of float
+        """
+        motorIndexes = [0, 1, 2, 3, 4, 5]
+
+        for i, ac in zip(motorIndexes, action):
+            self.motorPositionArr[i] += ac
+            self.motorList[i].setVelocity(self.motor_velocity)
+            self.motorList[i].setPosition(ac)
+
+    def setup_agent(self):
+        """
+        This method initializes the 17 (all) or 6 (leg)  motors, 
+        storing the references inside a list and setting the starting
+        positions and velocities.
+        """
+        self.motorList = RobotFunc.getAllMotors(self)
```

Comment:
During examining all the code, i arrived to the conclusion that maybe we need  to fully get rid of the utils file and just put the functionality in this method here. It is not actually that complicated to warrant the use of a different method.

Reply:
As long as the normalization function is not needed of course.

Reply:
Also something i missed is that `self.motorList` produces `Instance attribute motorList defined outside __init__ `

Reply:
> During examining all the code, i arrived to the conclusion that maybe we need to fully get rid of the utils file and just put the functionality in this method here. It is not actually that complicated to warrant the use of a different method.

I also create a utils file, `ArmUtils.py`, for panda since these functions is a little bit messy. Also, different controllers can access the same utils file.
* https://github.com/aidudezzz/deepworlds/pull/42#pullrequestreview-672039857

Reply:
> As long as the normalization function is not needed of course.

I think we need `normalizeToRange()` to fix the [SB3 issues]( https://github.com/aidudezzz/deepworlds/pull/76#issuecomment-1175738193)
* UserWarning: We recommend you to use a symmetric and normalized Box action space (range=[-1, 1])

Reply:
> I also create a utils file, ArmUtils.py, for panda since these functions is a little bit messy. Also, different controllers can access the same utils file.

The panda example seems to have more util functions and more complicated at that and two examples, so it's good to have a common util file. For this example, i think it boils down to putting 4 additional lines of motor initialization in the environment (setup_agent):

```python
  def setup_agent(self):
      self.motorList = RobotFunc.getAllMotors(self)
```
->

```python
  def setup_agent(self):
      motorNames = ['LeftAnkle', 'LeftCrus', 'LeftFemur',
                      'RightAnkle', 'RightCrus', 'RightFemur']
        # motorNames = ['Head', 'LeftAnkle', 'LeftArm', 'LeftCrus',
        #               'LeftElbow', 'LeftFemur', 'LeftFemurHead1',
        #               'LeftFemurHead2', 'LeftFoot', 'LeftForearm',
        #               'LeftShoulder', 'RightAnkle', 'RightArm',
        #               'RightCrus', 'RightElbow', 'RightFemur',
        #               'RightFemurHead1', 'RightFemurHead2',
        #               'RightFoot', 'RightForearm', 'RightShoulder',
        #               'Waist']

        self.motorList = []
        for motorName in motorNames:
            motor = self.getDevice(motorName)  # Get the motor handle
            motor.setPosition(float('inf'))  # Set starting position
            motor.setVelocity(0.0)  # Zero out starting velocity
            self.motorList.append(motor)  # Append motor to motorList
```

and by that we can get rid of two functions altogether making the code more transparent, not that different from [cartpole](https://github.com/aidudezzz/deepworlds/blob/5e815cf829eea46e05e82d8819cc3d755b9f909a/examples/cartpole/cartpole_discrete/controllers/robot_supervisor_manager/robot_supervisor.py#L177-L188). If for whatever reason these two functions were used elsewhere, i would suggest to keep it as it is.

> I think we need `normalizeToRange()` to fix the [SB3 issues](https://github.com/aidudezzz/deepworlds/pull/76#issuecomment-1175738193)

 Yes that's true. So we keep it and use it when we fix the convergence issues!

Reply:
>  For this example, i think it boils down to putting some additional lines of motor initialization in the environment (setup_agent) and by that we can get rid of two functions altogether making the code more transparent, not that different from cartpole.

@tsampazk Okay, I will fix that!